src/components/Table/TableSaveViewModal/TableSaveViewModal.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+import ComposedModal from '../../ComposedModal/ComposedModal';
+import { settings } from '../../../constants/Settings';
+import { OverridePropTypes } from '../../../constants/SharedPropTypes';
+
+import TableSaveViewForm from './TableSaveViewForm';
+
+const { iotPrefix } = settings;
+
+const propTypes = {
+  /**
+   * Callbacks for the actions of the modal
+   * onSave : Called with { title: string, defaultView: boolean, publicView: boolean} on save button click
+   * onClose : Called on cancel button click and on the top right close icon click
+   * onClearError : Called when the error msg is cleared
+   * onChange : Called when view title input value is changed
+   */
+  actions: PropTypes.shape({
+    onSave: PropTypes.func,
+    onClose: PropTypes.func,
+    onClearError: PropTypes.func,
+    onChange: PropTypes.func,
+  }).isRequired,
+  /** Shows this string as a general modal error when present */
+  error: PropTypes.string,
```

Comment:
This might be better suited to live in the `i18n` object since it will always be some sort of internationalized/translated string.

Reply:
Yes, I considered that, but other components, like PageWizard and StatefulPageWizard that have both and error prop and an i18n object prop, keep them separated. I think that is because the error prop is also slightly different than normal i18n props since the error prop is also the trigger to make the error appear.